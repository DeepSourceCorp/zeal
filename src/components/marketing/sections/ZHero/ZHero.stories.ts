import ZHero from './ZHero.vue'

import { action } from '@storybook/addon-actions'

import '../../../assets/css/tailwind.css'
import '../../../assets/css/typography.css'
import '../../../assets/css/layout.css'

export default {
  title: 'Zeal/Marketing/Sections/Hero',
  component: ZHero,
  excludeStories: /.*Data$/
}

export const VerticalItems = () => ({
  components: { ZHero },
  template: `
    <z-hero direction="something">
      <h1>Hero Title</h1>
      <h2>Hero Subtitle</h2>
      <p>This is a paragraph</p>
      <img slot="illustration" src="https://i.imgur.com/C3IjDqe.jpg" />
    </z-hero>
  `
})
