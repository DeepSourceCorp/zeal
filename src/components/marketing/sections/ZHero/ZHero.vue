<template>
  <section class="z-hero__section w-full relative text-vanilla-100 text-center">
    <span class="z-hero__content block">
      <h1
        class="block text-vanilla-100 mb-3 sm:mb-6 text-2xl sm:text-3xl lg:text-5xl leading-10 sm:leading-11 lg:leading-12 w-auto lg:w-4/5 font-bold mx-auto sm:px-10 md:px-5"
      >
        <slot name="heading"></slot>
      </h1>
      <h2
        class="block text-sm sm:text-lg lg:text-lg leading-6 sm:leading-9 lg:leading-normal w-auto lg:w-1/2 font-normal text-vanilla-200 mb-12 mx-auto px-5 sm:px-32 md:px-16"
      >
        <slot name="subheading"></slot>
      </h2>
      <span class="block">
        <slot name="cta"></slot>
      </span>
    </span>

    <div class="z-hero-illustration__content mt-10 lg:-mt-12">
      <!-- This div is for scaling from a (a < 1) to b (a < b <= 1) -->
      <div class="z-hero-illustration__container">
        <!-- This div for rotating on the z-axis from `a`deg (a > 0deg) to `b`deg ( 0deg <= b < a) -->
        <slot name="illustration"></slot>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import Vue from 'vue'

const MIN_SCALE = 0.7
const MAX_SCALE = 0.95
const MIN_ROTATION_ANGLE = 0
const MAX_ROTATION_ANGLE = 20

export default Vue.extend({
  name: 'ZHero',
  mounted() {
    this.handleScroll()
    document.addEventListener('scroll', this.handleScroll)
  },
  beforeDestroy() {
    document.removeEventListener('scroll', this.handleScroll)
  },
  methods: {
    handleScroll() {
      if (window.innerWidth > 833) {
        this.changeOpacity()
      }
      this.scaleIllustration()
      this.rotateIllustration()
    },
    changeOpacity() {
      const heroContent = document.getElementsByClassName('z-hero__content')[0] as HTMLElement
      heroContent.style.opacity = `${1 - window.scrollY / 400}`
    },
    scaleIllustration() {
      const illustrationContent = document.getElementsByClassName(
        'z-hero-illustration__content'
      )[0] as HTMLElement
      illustrationContent.style.transform = `
        translateX(0px)
        scale(${this.getScaleValue(window.scrollY)}) translateZ(0px)`
    },
    rotateIllustration() {
      const illustrationContainer = document.getElementsByClassName(
        'z-hero-illustration__container'
      )[0] as HTMLElement
      illustrationContainer.style.transform = `
        perspective(800px)
        rotateX(${this.getRotationValue(window.scrollY)}deg)
        translateZ(0px)`
    },
    getScaleValue(scrollY: number): number {
      const calc = MIN_SCALE + (scrollY * 0.08) / 100
      return calc < MAX_SCALE ? calc : MAX_SCALE
    },
    getRotationValue(scrollY: number): number {
      const calc = MAX_ROTATION_ANGLE - (scrollY * 0.6) / 10
      return calc > 0 ? calc : MIN_ROTATION_ANGLE
    }
  }
})
</script>
