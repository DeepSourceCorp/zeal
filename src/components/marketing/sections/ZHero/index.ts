import ZHero from './ZHero.vue'

export default ZHero
