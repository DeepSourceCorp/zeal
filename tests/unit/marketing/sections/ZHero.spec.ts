import Vue from 'vue'
import { createLocalVue, mount, shallowMount, Wrapper } from '@vue/test-utils'

import ZHero from '../../../../src/components/marketing/sections/ZHero'

describe('Customers section', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZHero, {
        localVue,
        ...options
      })
    }
  })
  it('renders hero section when the component is mounted', () => {
    let mountedInstance = mountFunction()
    expect(mountedInstance.html()).toMatchSnapshot()
  })

  it('renders the slots of hero section', () => {
    let mountedInstance = mountFunction({
      slots: {
        cta: '<span class="hero-cta">This is cta</span>',
        heading: '<span class="hero-heading">This is heading</span>',
        subheading: '<span class="hero-subheading">This is subheading</span>',
        illustration: '<span class="hero-illustration"><img src="https://www.tibco.com/blog/wp-content/uploads/2016/07/rsz_bigstock-electronic-devices-with-blank-81878129.jpg" /></span>'
      }
    })
    expect(mountedInstance.html()).toContain('hero-cta')
    expect(mountedInstance.html()).toContain('hero-heading')
    expect(mountedInstance.html()).toContain('hero-subheading')
    expect(mountedInstance.html()).toContain('hero-illustration')
  })
})
