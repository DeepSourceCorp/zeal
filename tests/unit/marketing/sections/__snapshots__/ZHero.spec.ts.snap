// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customers section renders hero section when the component is mounted 1`] = `
<section class="z-hero__section w-full relative text-vanilla-100 text-center"><span class="z-hero__content block" style="opacity: 1;"><h1 class="block text-vanilla-100 mb-4 sm:mb-6 text-2xl xs:text-2.5xl sm:text-3xl lg:text-5xl leading-10 sm:leading-11 lg:leading-12 w-auto lg:w-4/5 font-bold mx-auto sm:px-20"></h1> <h2 class="block text-sm sm:text-lg lg:text-lg leading-6 sm:leading-9 lg:leading-normal w-auto lg:w-1/2 font-normal text-vanilla-200 mb-12 mx-auto px-5 xs:px-12 sm:px-40 lg:px-16"></h2> <span class="block"></span></span>
  <div class="z-hero-illustration__content mt-10 lg:-mt-16" style="transform: 
        translateX(0px)
        scale(0.7) translateZ(0px);">
    <div class="z-hero-illustration__container" style="transform: 
        perspective(800px)
        rotateX(20deg)
        translateZ(0px);"></div>
  </div>
</section>
`;
